src/spatial/motion_force_interaction.jl

```diff
@@ -35,22 +35,36 @@ end
 
 # SpatialInertia-specific functions
 Base.eltype(::Type{SpatialInertia{T}}) where {T} = T
-@inline Base.convert(::SpatialInertia{T}, inertia::SpatialInertia{T}) where {T} = inertia
-@inline function Base.convert(::Type{SpatialInertia{T}}, inertia::SpatialInertia) where {T}
-    SpatialInertia(inertia.frame,
-        convert(SMatrix{3, 3, T}, inertia.moment),
-        convert(SVector{3, T}, inertia.cross_part),
-        convert(T, inertia.mass))
+
+# Construct/convert given another SpatialInertia
+SpatialInertia{T}(inertia::SpatialInertia{T}) where {T} = inertia
```

Comment:
Is this no-op constructor actually necessary?

Reply:
I think so. Before https://github.com/JuliaRobotics/RigidBodyDynamics.jl/pull/498/commits/6f15fe5669839fda157c2daf8f608a0ddc941a7a, deleting this line resulted in an error. After that commit, I checked the `code_llvm` and benchmark times with and without this line.

Setup code:

```julia
using RigidBodyDynamics, BenchmarkTools
f = CartesianFrame3D()
I = rand(SpatialInertia{Float64}, f)
```

With that line:

```julia
julia> @code_native SpatialInertia{Float64}(I)
        .text
; Function Type {
; Location: motion_force_interaction.jl:40
        vmovups (%rdx), %ymm0
        vmovups 32(%rdx), %ymm1
        vmovups 64(%rdx), %ymm2
        vmovups 80(%rdx), %ymm3
        vmovups %ymm3, 80(%rdi)
        vmovups %ymm2, 64(%rdi)
        vmovups %ymm1, 32(%rdi)
        vmovups %ymm0, (%rdi)
        movq    %rdi, %rax
        vzeroupper
        retq
        nopl    (%rax)
;}
```

Without that line:

```julia
julia> @code_native SpatialInertia{Float64}(I)
        .text
; Function Type {
; Location: motion_force_interaction.jl:42
; Function Type; {
; Location: motion_force_interaction.jl:42
        movq    (%rdx), %rax
;}
        movq    %rax, (%rdi)
        movq    72(%rdx), %rax
        movq    %rax, 72(%rdi)
        vmovups 8(%rdx), %ymm0
        vmovups 40(%rdx), %ymm1
        vmovups %ymm1, 40(%rdi)
        vmovups %ymm0, 8(%rdi)
        movq    96(%rdx), %rax
        movq    %rax, 96(%rdi)
        vmovups 80(%rdx), %xmm0
        vmovups %xmm0, 80(%rdi)
; Function Type; {
; Location: motion_force_interaction.jl:24
        movq    104(%rdx), %rax
;}
        movq    %rax, 104(%rdi)
        movq    %rdi, %rax
        vzeroupper
        retq
        nopw    %cs:(%rax,%rax)
;}
```

I.e., a bunch more `movq`s. Also, I think this is better for constant-folding, since:

With that line: 
```julia
julia> @btime SpatialInertia{Float64}($I)
  0.017 ns (0 allocations: 0 bytes)
```

Without that line:

```julia
julia> @btime SpatialInertia{Float64}($I)
  2.024 ns (0 allocations: 0 bytes)
```

However, strangely:

With that line:

```julia
julia> @btime SpatialInertia{Float64}(I) setup = I = rand(SpatialInertia{Float64}, $f);
  2.667 ns (0 allocations: 0 bytes)
```

Without that line:
```julia
julia> @btime SpatialInertia{Float64}(I) setup = I = rand(SpatialInertia{Float64}, $f);
  2.379 ns (0 allocations: 0 bytes)
```

But I think that might just be a microbenchmark artifact, even though it's reliably reproducable.

Reply:
Huh, fascinating